spec/yelp/burst_struct_spec.rb

```diff
@@ -1,155 +0,0 @@
-require 'yelp/burst_struct'
-
-describe BurstStruct::Burst do
-  describe '#keys' do
-    subject(:struct) { BurstStruct::Burst.new(foo: 'bar', baz: 'qux') }
-
-    it 'should return' do
-      expect(struct.keys).to eql [:foo, :baz]
-    end
-  end
-
-  describe '#foo' do
-    subject(:struct) { BurstStruct::Burst.new(foo: 'bar') }
-
-    context 'when a key exists' do
-      it 'should return' do
-        expect(struct.foo).to eql 'bar'
-      end
-
-      it { should have_key(:foo) }
-      it { should have_key('foo') }
-
-      it { should respond_to(:foo) }
-    end
-
-    context 'when a key does not exist' do
-      it 'should not respond to it' do
-        expect(struct.respond_to? :super_foo).to eql false
-      end
-
-      it { should_not have_key(:super_foo) }
-      it { should_not have_key('super_foo') }
-    end
-  end
-
-  context 'n deep nested hash' do
-    let(:hash) { Hash[ futurama: { characters: { robots: { best: 'bender' } } } ] }
-
-    subject(:struct) { BurstStruct::Burst.new(hash) }
-
-    describe '#foo#bar#baz#biz' do
-      it 'should return' do
-        expect(struct.futurama.characters.robots.best).to eql 'bender'
-      end
-    end
-  end
-
-  context 'arrays' do
-    let(:hash) { Hash[ businesses: [ { name: 'Yelp', location: 'San Francisco' },
-                                     { name: 'Ice Cream', flavor: 'Chocolate' },
-                                     [ { name: 'Moe', occupation: 'Bartender' } ] ] ] }
-
-    subject(:struct) { BurstStruct::Burst.new(hash) }
-
-    describe '#businesses[0].name' do
-      it 'should return' do
-        expect(struct.businesses[0].name).to eql 'Yelp'
-        expect(struct.businesses[0].location).to eql 'San Francisco'
-      end
-    end
-
-    describe '#business[1].name' do
-      it 'should return' do
-        expect(struct.businesses[1].name).to eql 'Ice Cream'
-        expect(struct.businesses[1].flavor).to eql 'Chocolate'
-      end
-    end
-
-    describe 'nested arrays' do
-      it 'should parse arrays all the way down' do
```

Comment:
Nope! This entire spec file was related to the class I had that would dynamically parse and create objects based on the response field. Given the complexity and a ton of metaprogramming involved, it definitely needed a lot of complex tests.

Right now, all of the new objects don't do anything special. They inherit from a `Base` class that does some very basic parsing, and that's about it. The base class has two tests to cover the basic parsing and variable creation it does, but outside of that I didn't see a ton of value in creating a lot of repetitive tests that verify that the base class is doing what it's supposed to be (which is tested).

Happy to discuss this more though if you don't agree!